### PR TITLE
 space-travelers-hub complete page

### PR DESCRIPTION
**Pull request by TamaraDimov**
Setup the project with basic dependencies, navigation bar, add empty profile page.
Create basic structure for rockets,missions. create redux store , rocket and mission redux slices.
Fetch rockets and missions data from the SpaceX API. Rockets fetched only once when App is rendered. missions fetched only once when user navigates to the missions page
Display rockets list in rockets page and missions list in missions page.
Implement rockets and missions booking functionality.
Implement rockets and missions cancel booking functionality.
Display conditionally , reserved badge for reserved rockets. Not display reserved badges for unreserved rockets.
Display leave mission badge for joined missions, display not a member badge for missions not joined.
Add unit test cases
Display reserved rockets list and joined missions in the profile page.
Document the project in the project readme

### CODE REVIEW

File: src/components/Mission/ProfileMissions.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+export default function ProfileMissions() {
+  const { missionList } = useSelector((store) => store.missions);
+  const joinedMission = missionList.filter(
+    (mission) => mission.reserved === true,
+  );
+  if (joinedMission.length === 0) {
```
Comment (topeogunleye) [inline src/components/Mission/ProfileMissions.js]: - [ ] As required by project requirements, please kindly make use of React conditional rendering syntax in your Rockets and Missions.

### For Example:

```
      <td>
        {mission.reserved && (
          <div className="status w-64 flex justify-around text-xs">
            <span className="badge bg-blue-400 text-white rounded-sm px-1 py-0.5 uppercase grid place-items-center">
              Active Member
            </span>
          </div>
        )}
        {!mission.reserved && (
          <div className="status w-64 flex justify-around text-xs">
            <span
              type="button"
              className="badge bg-gray-500 text-white rounded-sm uppercase grid place-items-center px-1 py-0.5"
            >
              not a member
            </span>
          </div>
        )}
      </td>
      <td>
        {mission.reserved && (
          <div className="w-64 flex justify-around text-xs">
            <button
              type="button"
              className="text-sm text-red-600 border-red-400 rounded-sm border-2 px-1 py-0.5"
              onClick={() => dispatch(leaveMission(mission.mission_id))}
            >
              {' '}
              leave mission
            </button>
          </div>
        )}
        {!mission.reserved && (
          <div className="w-64 flex justify-around text-xs">
            <button
              type="button"
              className="text-sm text-gray-600 border-gray-400 rounded-sm border-2 px-1 py-0.5"
              onClick={() => dispatch(joinMission(mission.mission_id))}
            >
              {' '}
              Join Mission
            </button>
          </div>
        )}
      </td>
  ));
```



The above is an example and may not apply directly to your code.

File: src/components/Rocket/ProfileRockets.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+export default function ProfileRockets() {
```
Comment (topeogunleye) [inline src/components/Rocket/ProfileRockets.js:4]: - [x] As required by project requirements, please kindly make use of React conditional rendering syntax in your Rockets and Missions.

### For Example:


  ```
 <span>
          {rockets.reserved && (
              <span className="inline-flex items-center justify-center px-2 py-1 mr-2 text-xs font-bold  text-white bg-[#18A2B8] rounded-md">Reserved</span>
              )}
            </span>
            <div className="mb-2">{rockets.description}</div>
            <div className="text-container">
              {rockets.reserved && (
              <button className="   bg-white p-1 text-gray-500 border-2 rounded-sm hover:bg-gray-200" 
                    type="button" onClick={() => dispatch(canRocket(rockets.id))}>
                {' '}
                Cancel Reservation
              </button>
              )}
              {' '}
              {!rockets.reserved && (
              <button className="rounded-sm px-2 p-1 bg-blue-500 text-white" type="button" onClick={() => 
                   dispatch(resRocket(rockets.id))}>
                {' '}
                Reserve Rocket
              </button>
```

The above is an example and may not apply directly to your code

File: src/components/Navigation.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  return (
+    <>
+      <div className={style.app}>
+        <nav className="navContainer">
+          <div className="title">
+            <img src={planet} alt="Space Travel Hub logo" className="logo" />
+            <h1 className="navHeader">Space Traveler&apos;s Hub</h1>
+          </div>
+          <ul className="navLinks">
+            <li>
```
Comment (topeogunleye) [inline src/components/Navigation.js:15]: - [x] I kindly suggest that you make use of `<NavLink>` for navigation as indicated in the requirements. Also kindly make sure that only the active page is indicated with an underline. With Navlink you can make your active link underlined very easily:
image

## Example of Navlink:

```
 <NavLink
      to="/rockets"
      style={({ isActive }) => ({
        textDecoration: isActive ? 'underline' : '#545e6f',
      })}
      className="navRockets"
    >
      {item.text}
  </NavLink>
```

The above is only an example and may not apply directly to your code.

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @TamaraDimov, @otmanTR, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
> * Good commit messages. :heavy_check_mark:
> * No linter errors. :heavy_check_mark:
> * Followed Github flow :heavy_check_mark:
> * Professional README file. :heavy_check_mark:
> * PR has a good title and summary :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @TamaraDimov and @otmanTR 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- All changes requested have been implemented   ✅ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Navigation.js (hunk 3 above)
Comment (otmanTR) [reply src/components/Navigation.js:15]: Thank you for these good suggestions.

Comment (topeogunleye) [reply src/components/Navigation.js:15]: @otmanTR you're welcome

**Pull request merged**